R/pkg/R/functions.R

```diff
@@ -2638,3 +2638,81 @@ setMethod("sort_array",
             jc <- callJStatic("org.apache.spark.sql.functions", "sort_array", x@jc, asc)
             column(jc)
           })
+
+#' This function computes a histogram for a given SparkR Column.
+#' 
+#' @name histogram
+#' @title Histogram
+#' @param nbins the number of bins (optional). The default is 10.
+#' @param df the DataFrame containing the Column to build the histogram from.
+#' @param colname the name of the column to build the histogram from.
+#' @return a data.frame with the histogram statistics, i.e., counts and centroids.
+#' @examples \dontrun{
+#' 
+#' # Create a DataFrame from the Iris dataset
+#' irisDF <- createDataFrame(sqlContext, iris)
+#' 
+#' # Compute histogram statistics
+#' histData <- histogram(df, "colname"Sepal_Length", nbins = 12)
+#'
+#' # Once SparkR has computed the histogram statistics, it would be very easy to
+#' # render the histogram using R's visualization packages such as ggplot2.
+#'   
+#' } 
+setMethod("histogram",
+          signature(df = "DataFrame"),
+          function(df, colname, nbins = 10) {
+            # Validate nbins
+            if (nbins < 2) {
+              stop("The number of bins must be a positive integer number greater than 1.")
```

Comment:
Yup, you could have a histogram with 2 bins:

> histogram(irisDF, "Sepal_Length", nbins=2)
bins counts centroids
1    0     95       5.2
2    1     55       7.0